shop/src/main/java/ro/msg/learning/shop/service/StockService.java

```diff
@@ -0,0 +1,46 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.entity.Stock;
+import ro.msg.learning.shop.repo.StockRepo;
+import ro.msg.learning.shop.utils.ID;
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+@RequiredArgsConstructor
+public class StockService {
+    private final StockRepo stockRepository;
+    private final EntityManager entityManager;
+
+    public void updateStock(Stock stockToUpdate, Integer quantityTaken){
+        Integer newQuantity= stockToUpdate.getQuantity()-quantityTaken;
+        stockToUpdate.setQuantity(newQuantity);
+        stockRepository.save(stockToUpdate);
+    }
+    public List<Stock> findStocksByLocationAndProductIDs(List<ID> locationAndProductIDsList) {
+        CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+        CriteriaQuery<Stock> query = cb.createQuery(Stock.class);
+        Root<Stock> stock = query.from(Stock.class);
+        Path<Integer> locationIdPath = stock.get("location").get("id");
+        Path<Integer> productIdPath = stock.get("product").get("id");
+        List<Predicate> predicates = new ArrayList<>();
+
+        locationAndProductIDsList.forEach(locationProductPair ->
+                predicates.add(
+                        cb.and(
```

Comment:
cb not a good name. Why? at any moment when someone looks over the code they should be able to know what happens there and this is made much easier by naming things the right way: as descriptive as possible